pkg/helm/watcher/cache/cache.go

```diff
@@ -158,6 +160,46 @@ func (c *ProfileCache) ListProfiles(ctx context.Context, helmRepoNamespace, helm
 	return result, nil
 }
 
+// ListAvailableVersionsForProfile returns all stored available versions for a profile.
+func (c *ProfileCache) ListAvailableVersionsForProfile(ctx context.Context, helmRepoNamespace, helmRepoName, profileName string) ([]string, error) {
+	// Because the folders of versions are only stored when there are values for a version
+	// we, instead, look in the profiles.yaml file for available versions.
+	var result []string
+
+	getAllAvailableVersionsOp := func() error {
+		var profiles []*pb.Profile
+
+		content, err := os.ReadFile(filepath.Join(c.cacheLocation, helmRepoNamespace, helmRepoName, profileFilename))
+		if err != nil {
+			// don't process if it doesn't exist.
+			if os.IsNotExist(err) {
+				return nil
+			}
+
+			return fmt.Errorf("failed to read profiles data for helm repo: %w", err)
+		}
+
+		if err := yaml.Unmarshal(content, &profiles); err != nil {
+			return fmt.Errorf("failed to unmarshal profiles data: %w", err)
+		}
```

Comment:
this part is probably repeated in other funcs too right? worth extracting out into a `getProfilesFromFile` func or something?

Reply:
Yeah, now it is. It wasn't previously. So yes, maybe it's worth extracting it now.

Reply:
Though, it varies slightly. Here, we ignore if the file is not found.